node/core/pvf/src/worker_common.rs

```diff
@@ -357,25 +338,35 @@ impl futures::Future for WorkerHandle {
 
 	fn poll(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
 		let me = self.project();
-		match futures::ready!(AsyncRead::poll_read(me.stdout, cx, &mut *me.drop_box)) {
-			Ok(0) => {
-				// 0 means `EOF` means the child was terminated. Resolve.
-				Poll::Ready(())
-			},
-			Ok(_bytes_read) => {
-				// weird, we've read something. Pretend that never happened and reschedule ourselves.
-				cx.waker().wake_by_ref();
-				Poll::Pending
+		// Create a `ReadBuf` here instead of storing it in `WorkerHandle` to avoid a lifetime
+		// parameter on `WorkerHandle`. Creating the `ReadBuf` is fairly cheap.
+		let mut read_buf = ReadBuf::new(&mut *me.drop_box);
+		match futures::ready!(AsyncRead::poll_read(me.stdout, cx, &mut read_buf)) {
+			Ok(()) => {
+				if read_buf.filled().len() > 0 {
+					// weird, we've read something. Pretend that never happened and reschedule
+					// ourselves.
+					cx.waker().wake_by_ref();
+					Poll::Pending
+				} else {
+					// Nothing read means `EOF` means the child was terminated. Resolve.
+					Poll::Ready(())
+				}
 			},
 			Err(err) => {
 				// The implementation is guaranteed to not to return `WouldBlock` and Interrupted. This
 				// leaves us with legit errors which we suppose were due to termination.
 
 				// Log the status code.
+				let code = if let Ok(Some(code)) = me.child.try_wait() {
+					format!("{}", code)
+				} else {
+					"none".into()
+				};
 				gum::debug!(
 					target: LOG_TARGET,
-					worker_pid = %me.child.id(),
-					status_code = ?me.child.try_status(),
+					worker_pid = %me.child_id,
+					status_code = ?code,
```

Comment:
```suggestion
					status_code = ?me.child.try_wait().ok().flatten().map(|c| c.to_string()),
```

We only need to call this code when `debug` logs are enabled.